### DISCUSSION TITLE
Closeness centrality that considers unreachable nodes (Networkx vs SAS discrepancy)

### DISCUSSION THREAD

**Discussion by igygi** [networkx/networkx#4933] [Algorithms]
Hi, 

I'm comparing networkx's implementation for closeness centrality with [SAS](https://documentation.sas.com/doc/en/procgralgcdc/14.2/procgralg/procgralg_optgraph_details17.htm#procgralg.optgraph.optgr_centr_close). And it seems they vary in the nodes being considered in the computation. 

In SAS, unreachable nodes are included in the computation and they are assigned a dummy shortest path metric `p` that could be derived in various ways (e.g. graph diameter). 
![image](https://user-images.githubusercontent.com/43819944/123585939-96740c80-d816-11eb-9e3b-d95a49ae04cd.png)

While in networkx, only reachables nodes are considered based on the [documentation](https://networkx.org/documentation/stable/_modules/networkx/algorithms/centrality/closeness.html#closeness_centrality):

>     Closeness centrality [1]_ of a node `u` is the reciprocal of the
>     average shortest path distance to `u` over all `n-1` reachable nodes.
> 

Would you consider expanding networkx's implementation to support this alternative approach?

**Comment by BradKML** [comment]
Are there other centrality measures that also have this issue?
